src/utp_stream.cpp

```diff
@@ -3408,6 +3420,22 @@ bool utp_socket_impl::incoming_packet(boost::uint8_t const* buf, int size
 	return true;
 }
 
+boost::int32_t utp_socket_impl::adjust_receive_buf_size(boost::uint32_t const& min_rtt)
+{
+	// maximum value for RTT we would still scale for
+	const boost::int32_t rtt_ceil = 800000;
+
+	// linear buffer increase coefficient. Defines how fast
+	// m_in_buf_size would increase/decrease with increase/decrease of minimum RTT
+	const boost::int64_t k = (boost::int64_t(m_in_buf_size_max - m_in_buf_size_min)<<16) / rtt_ceil;
+
+	//calculate new size of the buffer and clamp it to max allowed value
+	const boost::int32_t adjusted_buffer_size = m_in_buf_size_min + boost::int32_t((min_rtt * k)>>16);
+	const boost::int32_t clamped_buffer_size = std::min(adjusted_buffer_size, int(m_in_buf_size_max));
+
+	return size_t(clamped_buffer_size);
```

Comment:
I think the ``size_t`` conversion here was a mistake. ``clamped_buffer_size`` already has the right type